### PR TITLE
Add more rules to ruff and apply them

### PR DESCRIPTION
**Pull request by lkstrp**
See #899 (Step 2 and 3)

## Checklist

- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,51 +3,30 @@ repos:
...
-
-# Formatting with "black" coding style
-- repo: https://github.com/psf/black
-  rev: 24.4.2
-  hooks:
-  # Format Python files
-  - id: black
-  # Format Jupyter Python notebooks
-  - id: black-jupyter
-
```
Comment (FabianHofmann) [inline .pre-commit-config.yaml:50]: that means ruff replaces docformatter, balck and isort?

**Review by FabianHofmann — APPROVED** [review]
Looks good! Feel free to merge :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (lkstrp) [reply .pre-commit-config.yaml:50]: Yes. For the Docstrings it covers mostly the same on what docformatter did and more (see [here](https://docs.astral.sh/ruff/rules/#pycodestyle-e-w)). I adjusted the rules in `pyproject.toml` and we can add some more in future (e.g. once we have docstrings everywhere, so it fails if you wanna merge something without docstring)

Comment (FabianHofmann) [reply .pre-commit-config.yaml:50]: great

**Pull request merged**